controllers/front/OrderDetailController.php

```diff
@@ -183,15 +183,27 @@ public function initContent()
 
             $order = new Order($id_order);
             if (Validate::isLoadedObject($order) && $order->id_customer == $this->context->customer->id) {
-                $this->order_to_display = (new OrderPresenter())->present($order);
+                $viewAccess = true;
```

Comment:
I see someone who will say "Deny first, and then allow when granted"
Ping @PierreRambaud

Reply:
I'm more in favor of failing fast.
Instead of having $viewAccess = false; I would prefer to directly redirect when it's possible.
Instead of having encapsulation